src/components/Templates/TemplatesList/FiltersSection.tsx

```diff
@@ -0,0 +1,111 @@
+import Accordion from '@components/Accordion';
+import type { Category, Filters, Framework, Template } from '../types';
+import Text from '@components/Text';
+import { FilterList } from './FilterList';
+import { useMemo } from 'react';
+
+type FilterItem = (Framework | Category) & { amount: number; avatar?: string };
+
+function groupByWithAmount<K extends 'framework' | 'category'>(
+  templates: Template[],
+  key: K,
+): (Template[K] & { amount: number })[] {
+  const map = new Map<string, Template[K] & { amount: number }>();
+
+  for (const template of templates) {
+    const item = template[key];
+    if (!item) continue;
+    const existing = map.get(item.name);
+    if (existing) {
+      existing.amount += 1;
+    } else {
+      map.set(item.name, { ...item, amount: 1 });
+    }
+  }
+
+  return Array.from(map.values());
+}
+
+const FilterLabel: React.FC<{ item: FilterItem }> = ({ item }) => {
+  const icon = item.avatar;
+
+  return (
+    <div className="ml-10 flex flex-row items-center justify-center">
+      {icon && (
+        <img
+          className="mr-10 inline-block h-16 w-16 rounded-full bg-neutral-6 lg:h-20 lg:w-20"
+          src={icon}
+          alt={item.name}
+        />
+      )}
+      <span className="text-[12px] lg:text-[14px]">{item.name}</span>
+      <span className="ml-10 h-16 w-16 rounded-full bg-neutral-4 text-center text-[10px] leading-16 lg:h-20 lg:w-20 lg:leading-20">
+        {item.amount}
+      </span>
+    </div>
+  );
+};
+
+interface FiltersSectionProps {
+  templates: Template[];
+  filters: Filters;
+  updateFilter: (
+    key: 'framework' | 'category' | 'search',
+    value: string | undefined,
```

Comment:
Where declared, has to assign undefined?

```suggestion
    value?: string,
```